### PR TITLE
Add static-analysis, align oss/base-poms more

### PR DESCRIPTION
**Pull request by nzroller**
* Makes osspom and basepom easier to diff

* Updates javadoc to 3.0.0 from 3.0.0-M1

* Fix dep-vuln-check without suppressions

  No longer forces projects to have a suppression file in order to run `-Pdep-vuln-check`.

* Adds `static-analysis` profile

* Adds `release-sign-artifacts` profile to basepom

  In preparation for signing internal artifacts too. The activation
  configuration is removed though so in order to use you will need to manually
  run with `-Prelease-sign-artifacts` ()or set activation in your settings.xml
  or profiles.xml).

* Add Jenkinsfile to build

* Add Dockerfile to build

  - Make build owner of m2 repository files, PATH includes mvn

### CODE REVIEW

**Comment by xiaodong-izettle** [comment]
Let me know when you would like this to be merged, @nzroller. 😄 

Nice work, 👍

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,6 @@
+@Library('japie@928f942') _
```
Comment (rickard-von-essen-iz) [inline Jenkinsfile]: Remove the hash.

**Comment by nzroller** [comment]
@xiaodong-izettle good to go man... we will fix CI after merge I reckon

**Comment by nzroller** [comment]
@rickard-von-essen-iz care to rereview?

**Pull request merged**